awx/main/tasks/jobs.py

```
@@ -1573,6 +1632,62 @@ def post_run_hook(self, instance, status):
             p.inventory_files = p.inventories
             p.save(update_fields=['scm_revision', 'playbook_files', 'inventory_files'])
 
+    def _copy_project_from_artifacts(self, instance):
+        """Copy project files from artifacts to projects_root after remote execution."""
+        artifacts_project_path = self.runner_callback.project_artifact_path
+
+        if not os.path.exists(artifacts_project_path):
+            raise PostRunError(
+                f'{instance.log_format} project artifacts not found after remote update: {artifacts_project_path}',
+                status='error',
+            )
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)